lib/unite_rb/scope.rb

```diff
@@ -3,16 +3,24 @@
 module UniteRb
   class Scope
     attr_reader :dims
+		def initialize
+			@dims ={}
+		end
+
+		def find_add_dimension(dim)
+			@dims[dim] ||  (@dims[dim] = Dimension.new(dim))
+		end
 
     def dimensions=(dims)
-      @dims = dims.each_with_object({}) do |d, acc|
+			@dims = dims.each_with_object({}) do |d, acc|
         acc[d] = Dimension.new(d)
       end
     end
 
-    def var(val, dim_name)
-      dim = @dims[dim_name]
-      raise UndefinedDimension.new("Dimension #{dim_name} not defined") if dim.nil?
+		def var(val = 0, dim_name)
```

Comment:
In case I need to start a variable and start working on it afterwards.

Reply:
The way `Var` is defined now, you can't change its value after it's initialized, since the `val` method is read-only. We could have `attr_accessor :var` instead, but the benefit is not clear.

And my problem with `0` is that it seems arbitrary. If I think of a just a variable like `s.var(:m)`, it even would look more to me like a `1m` than a `0m`.